meteor/server/migration/addMissingIdsForSelectFromColumns.ts

```diff
@@ -0,0 +1,203 @@
+	}
+	return tableToSelectFrom.find((row) => row[configManifestEntry.columnId] === configToVerifyUsesId)
+}
+
+function fetchShowStyleBase(blueprintId: BlueprintId): ShowStyleBase | undefined {
+	const showStyles = ShowStyleBases.find({
+		blueprintId,
+	}).fetch()
+	if (!showStyles || showStyles.length === 0) {
+		return undefined
+	}
+	return showStyles[0]
+}
+
+function createMigrationsForShowStylVariants(blueprints: BlueprintWithShowStyleConfigManifest[]): MigrationStepBase[] {
+	return blueprints
+		.map((blueprint) => {
+			const showStyleBases = ShowStyleBases.find({ blueprintId: blueprint.blueprintId }).fetch()
+			return showStyleBases.map((base) => {
+				const showStyleVariants = ShowStyleVariants.find({ showStyleBaseId: base._id }).fetch()
+				return showStyleVariants.map((variant) => {
+					return blueprint.showStyleConfigManifest.map((showStyleManifest) =>
+						createShowStyleVariantBlueprintConfigurationSelectFromColumnMigration(
+							blueprint.blueprintId,
+							showStyleManifest,
+							variant
+						)
+					)
+				})
+			})
+		})
+		.flat(3)
+}
+
+function createShowStyleVariantBlueprintConfigurationSelectFromColumnMigration(
+	blueprintId: BlueprintId,
+	configManifestEntry: ConfigManifestEntry,
+	variant: ShowStyleVariant
+): MigrationStepBase {
+	return {
+		id: `${blueprintId}.variant.${variant.name}.add.missing.id.to.select.from.column.for.${configManifestEntry.name}.${configManifestEntry.id}`,
+		canBeRunAutomatically: true,
+		validate: () => {
+			const entryToMapFrom = findEntryToMapFromForShowStyleVariant(blueprintId, configManifestEntry, variant._id)
+			// If we find an entry to map that means we have an id to map and should run the migration
+			return !!entryToMapFrom
+		},
+		migrate: () => {
+			const entryToMapFrom = findEntryToMapFromForShowStyleVariant(blueprintId, configManifestEntry, variant._id)
+			if (!entryToMapFrom) {
+				return
+			}
+			ShowStyleVariants.update(
+				{
+					_id: variant._id,
+				},
+				{
+					$set: {
+						[`blueprintConfig.${configManifestEntry.id}`]: {
+							value: entryToMapFrom._id,
+							label: entryToMapFrom[
+								(configManifestEntry as ConfigManifestEntrySelectFromColumn<false>).columnId
+							],
+						},
+					},
+				}
+			)
+		},
+	}
+}
+
+function findEntryToMapFromForShowStyleVariant(
+	blueprintId: BlueprintId,
+	configManifestEntry: ConfigManifestEntry,
+	showStyleVariantId: ShowStyleVariantId
+) {
+	const showStyle: ShowStyleBase | undefined = fetchShowStyleBase(blueprintId)
+	if (!showStyle) {
+		return undefined
+	}
+	// Fetching the ShowStyleVariant again to ensure we work on updated data. Needed for validation, among others.
+	const variant = fetchShowStyleVariant(showStyleVariantId)
+	if (!variant) {
+		return undefined
+	}
+	return findEntryToMapFrom(configManifestEntry, variant.blueprintConfig, showStyle.blueprintConfig)
+}
+
+function fetchShowStyleVariant(showStyleVariantId: ShowStyleVariantId): ShowStyleVariant | undefined {
```

Comment:
This function can be rewritten to `return ShowStyleVariants.findOne(showStyleVariantId)`.

Reply:
Updated